From 9b2b143090170b8a2d1d6012d705630364f3ad1d Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Thu, 29 Jun 2017 10:12:46 -0400
Subject: [PATCH] data: Revert dropping gdm-shell.session

The removal caused problems for Ubuntu 17.10 which does not
include gnome.session by default.

https://launchpad.net/bugs/1701243

https://bugzilla.gnome.org/show_bug.cgi?id=784340
---
 data/Makefile.am                         | 12 ++++++++++++
 data/dconf/defaults/00-upstream-settings |  3 +++
 data/gdm-shell.session.in                |  3 +++
 3 files changed, 18 insertions(+)
 create mode 100644 data/gdm-shell.session.in

diff --git a/data/Makefile.am b/data/Makefile.am
index f2875fcb..e2d3ebe8 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -86,9 +86,20 @@ gdm.schemas.in: $(srcdir)/gdm.schemas.in.in
 		-e 's,[@]sbindir[@],$(sbindir),g' \
 		<$(srcdir)/gdm.schemas.in.in >gdm.schemas.in
 
+
+EXTRA_DIST += gdm-shell.session.in
+
+gdm-shell.session: $(srcdir)/gdm-shell.session.in
+	sed	-e 's,[@]libexecdir[@],$(libexecdir),g' \
+		-e 's,[@]CHECK_ACCELERATED_DIR[@],$(CHECK_ACCELERATED_DIR),g' \
+		< $< > $@.tmp && mv $@.tmp $@
+
 localealiasdir = $(datadir)/gdm
 localealias_DATA = locale.alias
 
+sessiondir = $(datadir)/gnome-session/sessions
+session_DATA =  gdm-shell.session
+
 pam_redhat_files = \
 	pam-redhat/gdm-autologin.pam		\
 	pam-redhat/gdm-launch-environment.pam	\
@@ -178,6 +189,7 @@ CLEANFILES = 				\
 
 DISTCLEANFILES =			\
 	$(dbusconf_DATA)		\
+	gdm-shell.session		\
 	gdm.schemas			\
 	$(NULL)
 
diff --git a/data/dconf/defaults/00-upstream-settings b/data/dconf/defaults/00-upstream-settings
index c0b3cc66..eb8bc33c 100644
--- a/data/dconf/defaults/00-upstream-settings
+++ b/data/dconf/defaults/00-upstream-settings
@@ -25,6 +25,9 @@ disable-print-setup=true
 disable-save-to-disk=true
 disable-user-switching=true
 
+[org/gnome/desktop/session]
+session-name='gdm-shell'
+
 [org/gnome/desktop/sound]
 event-sounds=true
 
diff --git a/data/gdm-shell.session.in b/data/gdm-shell.session.in
new file mode 100644
index 00000000..278d8d43
--- /dev/null
+++ b/data/gdm-shell.session.in
@@ -0,0 +1,3 @@
+[GNOME Session]
+Name=Display Manager
+RequiredComponents=org.gnome.Shell;org.gnome.SettingsDaemon.A11yKeyboard;org.gnome.SettingsDaemon.A11ySettings;org.gnome.SettingsDaemon.Clipboard;org.gnome.SettingsDaemon.Color;org.gnome.SettingsDaemon.Datetime;org.gnome.SettingsDaemon.Housekeeping;org.gnome.SettingsDaemon.Keyboard;org.gnome.SettingsDaemon.MediaKeys;org.gnome.SettingsDaemon.Mouse;org.gnome.SettingsDaemon.Orientation;org.gnome.SettingsDaemon.Power;org.gnome.SettingsDaemon.PrintNotifications;org.gnome.SettingsDaemon.Rfkill;org.gnome.SettingsDaemon.ScreensaverProxy;org.gnome.SettingsDaemon.Sharing;org.gnome.SettingsDaemon.Smartcard;org.gnome.SettingsDaemon.Sound;org.gnome.SettingsDaemon.Wacom;org.gnome.SettingsDaemon.XRANDR;org.gnome.SettingsDaemon.XSettings;
-- 
2.11.0
